builtin/mainmenu/tab_settings.lua

```diff
@@ -203,9 +203,20 @@ local function formspec(tabview, name, tabdata)
 		"label[4.25,3.45;" .. fgettext("Screen:") .. "]" ..
 		"checkbox[4.25,3.6;cb_autosave_screensize;" .. fgettext("Autosave Screen Size") .. ";"
 				.. dump(core.settings:get_bool("autosave_screensize")) .. "]" ..
-		"box[8,0;3.75,4.5;#999999]" ..
-		"checkbox[8.25,0;cb_shaders;" .. fgettext("Shaders") .. ";"
-				.. dump(core.settings:get_bool("enable_shaders")) .. "]"
+		"box[8,0;3.75,4.5;#999999]"
+
+	local video_driver = core.settings:get("video_driver")
+	local shaders_supported = video_driver == "opengl"
+	local shaders_enabled = shaders_supported and core.settings:get_bool("enable_shaders")
+	if shaders_supported then
+		tab_string = tab_string ..
+			"checkbox[8.25,0;cb_shaders;" .. fgettext("Shaders") .. ";"
+					.. dump(core.settings:get_bool("enable_shaders")) .. "]"
```

Comment:
Using the `dump` output here is wrong (was already wrong before) because the output format may change any time. `tostring` might be more appropriate here.